server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/ComponentRootBuilder.java

```diff
@@ -73,9 +105,18 @@ private ComponentImpl buildComponent(ScannerReport.Component reportComponent, St
     }
   }
 
+  private ComponentImpl buildProjectComponent(ScannerReport.Component reportComponent, String latestModuleKey) {
+    ComponentImpl.Builder builder = createCommonBuilder(reportComponent, latestModuleKey, latestModuleKey);
+    return builder
+      .setName(nameOfProject(reportComponent, latestModuleKey, projectDtoSupplier))
+      .setReportAttributes(createProjectReportAttributes(reportComponent, builder.getUuid(), analysisSupplier))
```

Comment:
It's not that weird, especially considering builder as reusable (eg. protobuf generated builders).
I initially considered moving uuid generation out but it added both complexity (more parameters) and duplication